ASC: Updates (#2322)
ppy/osu-wiki: wiki/Article_Styling_Criteria/en.md

### Metadata

Metadata must be placed at the very top of the file. An empty line must follow it.

#### Outdated articles

*Note to translators: The `outdated` mark must be removed after updating the translation of an article.*

Translated articles that are outdated must use the `outdated` mark when the English variant is updated. The markup is as follows:

```md
---
outdated: true
---

# Beatmaps
```

#### Tagging articles

Tagging articles will help the website's search engine query articles better. Tags should use lowercase letters. It may be unlikely that this is needed for most articles; however, the markup is as follows:

```md
---

> tags:

Reviewer: Nit: I think since this is a front matter with a YAML syntax, we should have it as `yml` instead.

Author: Down below on line 87, I included a title to show that a space is required. Changing it to `yaml` will make that look like a comment instead. (keeping it as-is)

### Article file names

The file name of an article can be found in the `File Name` column of the [locales section](#locales). The location of a translated article must be placed in the same folder as the English version.

### Index articles

All folders should contain an article of some kind. If it seems impractical to create an article for a folder that holds other articles, create an index article instead. Index articles list off articles that are inside its own folder and may contain a description for each link.

### Disambiguation articles

[Disambiguation](/wiki/Disambiguation) articles must be placed in the `/wiki/Disambiguation` folder. The main page must be updated to include the disambiguation article. Refer to [Disambiguation/Mod](/wiki/Disambiguation/Mod) as an example.

Redirects must be updated to have the ambiguous keyword(s) redirect to the disambiguation article.

## Markdown

All osu! wiki articles must adhere to the following rules.

### GitHub Flavored Markdown

Some parts of the GitHub Flavored Markdown markup are supported in the osu! wiki.

### HTML

<!-- ppy ruled HTML is not allowed. from:ppy#0001 during:2017-06-26 in:osu-wiki markdown -->

> HTML must not be used, with exception (see section below). The structure of the article must be redone if HTML is used.

Reviewer: I think we should link the ASC section as-is.
```suggestion
using HTML is prohibited, with an exception for comments (see [comments](#comments)). The structure of the article must be redone if HTML is used.
```

- spaces
  - all spaces must be changed to underscores (`_`)
- apostrophes (`'`)

### Article file names

The file name of an article can be found in the `File Name` column of the [locales section](#locales). The location of a translated article must be placed in the same folder as the English version.

### Index articles

All folders should contain an article of some kind. If it seems impractical to create an article for a folder that holds other articles, create an index article instead. Index articles list off articles that are inside its own folder and may contain a description for each link.

### Disambiguation articles

[Disambiguation](/wiki/Disambiguation) articles must be placed in the `/wiki/Disambiguation` folder. The main page must be updated to include the disambiguation article. Refer to [Disambiguation/Mod](/wiki/Disambiguation/Mod) as an example.

Redirects must be updated to have the ambiguous keyword(s) redirect to the disambiguation article.

## Markdown

All osu! wiki articles must adhere to the following rules.

### GitHub Flavored Markdown

> Some parts of the GitHub Flavored Markdown markup are supported in the osu! wiki.

Reviewer: Let's make it a bit more formal-sounding.
```suggestion
A subset of the GitHub Flavored Markdown syntax are supported in the osu! wiki.
```

Reviewer: You might as also link this [article](https://help.github.com/en/articles/about-writing-and-formatting-on-github) about GFM on "GitHub Flavored Markdown" as well to give contributors an idea what GFM is.

Author: I forgot why I wrote that here. Does the website support any markup from GFM, other than the regular Markdown? A subset (or "some parts") of GFM seems vague or doesn't sound right if it is just Markdown.

I'll just remove the section instead, if there is no other discussion.

Author: Also, I think that article should be linked in the `osu! wiki contribution guide` instead.

Reviewer 2: https://github.github.com/gfm/

Tables are not supported in regular markdown, maybe that's why.

Reviewer: Markdown tables are part of the Markdown syntax spec @Joehuu

Reviewer: > Also, I think that article should be linked in the `osu! wiki contribution guide` instead.

good wiki practice is that this should also be linked in the criteria page as well. It's better you link GFM's definition since one reference on one article is not enough.

Reviewer 2: GFM as an extension, yes. Commonmark or regular markdown, no.

Reviewer 3: they aren't part of commonmark, which is the spec that osu! uses. osu! is also using other extensions and options that aren't standard though, so that's why these things are possible

anyway i think this note about GFM is a little pointless since any info it potentially gives is still meaningless (osu! has nothing to do with GFM, even if they happen to be similar). there's plenty of info about markdown in this article already

Reviewer: Yeah I think we should remove this altogether if that's the case.

### HTML

<!-- ppy ruled HTML is not allowed. from:ppy#0001 during:2017-06-26 in:osu-wiki markdown -->

HTML must not be used, with an exception for comments (see [comments](#comments)). The structure of the article must be redone if HTML is used.

#### Comments

HTML comments may be used for annotating text.

### Editing

#### End of line sequence

> <!-- Parsing will break if it reads a file using CRLF. -->

Reviewer 3: I don't think this is true, if you're referring to parsing on the website
not rly a problem though just a comment

Author: "Parsing" is broad, I could have worded that better. Probably something like: `<!-- YAML will display in the article on the website if the file is using CRLF -->`?

{artist} - {title} ({creator}) [{difficuty_name}]
```

> The link to a beatmap must actually link to that difficulty. All beatmap difficulty URLs looks like this: `https://osu.ppy.sh/beatmaps/{BeatmapID}`. The other URL format may be used instead.

Author: Should beatmap difficulty URLs be formatted like that? If so, `URLs should be` -> `URLs must be`.

> The link to a beatmap must actually link to that difficulty. All beatmap difficulty URLs looks like this: `https://osu.ppy.sh/beatmaps/{BeatmapID}`. The other URL format may be used instead.

Reviewer 2: Edited suggestion.
